### PR TITLE
ARC: fix issue with k_thread_create() handler checks for stack size

### PR DESCRIPTION
**Pull request by tyrantlizard77**
We were not accounting for the privileged stack area reserved, which is at the end of the stack buffer on this arch.

Fixes: #14643

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]
LGTM.

It might be a good idea to consolidate the #if in `include/arch/arm/arch.h` in the future.

**Review by andyross — APPROVED** [review]
Can't see anything wrong.  Does seem like we've got a lot of little "put this into the stack memory" layers that probably should be unified to a single API at some point.

**Review by vonhust — APPROVED** [review]
It looks ok to me as ARC put all stack elements (stack guard + user stack + privilege stack) statically together. So it's good to have a macro to let arch decides how much area is reserved.

**Comment by andrewboie** [comment]
Putting DNM tag on as I think I've found more bugs in this area.

**Review by agross-oss — APPROVED** [review]
lgtm

File: kernel/thread.c [hunk 1]
```diff
@@ -468,24 +466,18 @@ Z_SYSCALL_HANDLER(k_thread_create,
...
 							     &total_size),
 				    "stack size overflow (%u+%u)", stack_size,
-				    guard_size));
-#else
-	total_size = stack_size;
-#endif
-	/* They really ought to be equal, make this more strict? */
+				    K_THREAD_STACK_RESERVED));
+
+	/* Testing less-than-or-equal since additional room may have been
```
Comment (ioannisg) [inline kernel/thread.c]: We wouldn't have to do this @andrewboie if we were forced (by API) to always pass K_THREAD_STACK_SIZEOF(K_THREAD_STACK_DEFINE(sym)) to k_thread_stack_create.

**Pull request merged**